app/scripts/directives/bindService.js

```diff
@@ -195,13 +198,13 @@
 
     ctrl.bindService = function() {
       var svcToBind = ctrl.target.kind === 'Instance' ? ctrl.target : ctrl.serviceInstances[ctrl.serviceToBind];
-      var appToBind = ctrl.target.kind !== 'Instance' ? ctrl.target : ctrl.appToBind;
+      var application = ctrl.bindType === 'application' ? _.get(ctrl.appToBind, 'metadata.name') : undefined;
 
       var context = {
         namespace: _.get(svcToBind, 'metadata.namespace')
       };
 
-      BindingService.bindService(context, _.get(svcToBind, 'metadata.name'), _.get(appToBind, 'metadata.name')).then(function(binding){
+      BindingService.bindService(context, _.get(svcToBind, 'metadata.name'), application).then(function(binding){
```

Comment:
So DataService patterns don't lead with `context`, they tend towards something like `(resource, name, context, opts)`.  We missed this & should prob open an issue in common to keep these consistent.  I'm curious if `application` even ends up as part of context in the end, but I'd have to look.